dev-docs/workflows/build-develop-deploy.md

```diff
@@ -0,0 +1,180 @@
+
+## Prequisites
+
+### Linux
+
+* Packages on Ubuntu:
+
+  ```sh
+  sudo apt install build-essential cmake libssl-dev pkg-config libcryptsetup12 libcryptsetup-dev
+  ```
+
+* Packages on Fedora:
+
+  ```sh
+  sudo dnf install @development-tools pkg-config cmake openssl-devel cryptsetup-libs cryptsetup-devel
+  ```
+
+### Mac
+
+```sh
+brew install bash
+```
+
+to fix unsupported shell options used in some build script.
+
+To troubleshoot potential problems with bazel on ARM architecture when running it for the first time, it might help to purge and retry:
+
+```sh
+bazel clean --expunge
+```
+
+## Settings
+
+### Set build version
+
+There exists an `oss` and an `enterprise` version for licensing reasons.
+
+`enterprise`: recommended, complete feature set but some features are not licensed under Open-Source
+
+`oss`: default when building from the OSS repo, but non OSS-licensed features are missing (e.g. verify boot measurements). The [official docs](https://docs.edgeless.systems/constellation/) are written to be compatible with `oss`.
+<!-- are we sure about doc support? -->
+
+For convenience, we can set `enterprise` as the default build version:
+
+```
+cat .bazeloverwriterc
+build --cli_edition=enterprise
+```
+
+### Authenticate with the GitHub registry
+
+Get a GitHub token with access to the registry and authenticate through `docker` (see [here](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry)).
+
+Optionally, you can customize the default registry target:
+
+```sh
+cat .bazeloverwriterc
+build --container_prefix=ghcr.io/USERNAME/constellation
+```
+
+### Authenticate with your cloud provider
+
+To provision the constellation cluster on the provider infrastructure, please authenticate with respective provider.
+
+E.g. AWS:
+
+```sh
+aws login
+```
+
+For more details, see [here](https://docs.edgeless.systems/constellation/getting-started/install#set-up-cloud-credentials).
+
+## Build
+
+Build the binaries always in a separate directory:
+
+```sh
+mkdir build && cd build
+# build required binaries for a dev build
+# and symlink them into the current directory
+# also push the built container images
+# After the first run, set the pushed imaged to public.
+bazel run //:devbuild --cli_edition=oss --container_prefix=ghcr.io/USERNAME/constellation
+./constellation ...
+```
+
+### Specialized Bazel build targets
+
+see [bazel](./bazel.md#build)
+
+## Develop & Contribute
+
+### VS Code setup
+
+We recommend to set up your IDE to conform with our conventions (see [here](../conventions.md/#recommended-vs-code-settings)). You might also consider to set up Bazel in the IDE (see [here](./bazel.md#vs-code-integration)).
+
+### Testing
+
+See [here](./testing.md)
+
+### Pre-Opening a PR
+
+Before opening a PR, please run the tests and
+
+```sh
+bazel run //:generate && bazel run //:tidy
+bazel run //:check
+```
```

Comment:
```suggestion

Before opening a PR, please run the tests and regenerate all files:

´´´sh
bazel run //:generate && bazel run //:tidy
bazel run //:check
bazel test //...
´´´
```

We can easily include the bazel test target here.
(Also please ignore my backticks change again)